src/libstd/io/impls.rs

```diff
@@ -353,6 +362,9 @@ impl Write for Vec<u8> {
 
     #[inline]
     fn write_all(&mut self, buf: &[u8]) -> io::Result<()> {
```

Comment:
Now I'm thinking about this, maybe we should still write as much as possible, and thus fall back to the default `write_all` method. It's more consistent with the implementation of Write for slices.